cmd/openshift-install/create.go

```diff
@@ -262,8 +270,64 @@ func destroyBootstrap(ctx context.Context, directory string) (err error) {
 	return destroybootstrap.Destroy(rootOpts.dir)
 }
 
+// waitForconsole
+func waitForConsole(ctx context.Context, directory string) (string, error) {
+	url := ""
+	// Need to keep these updated if they change
+	consoleNamespace := "openshift-console"
+	consoleRouteName := "console"
+	config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(directory, "auth", "kubeconfig"))
```

Comment:
Instead of duplicating this in `destroyBootstrap` and `waitForConsole`, can we push it up into the caller?  With a few unrelated changes, that would be something like:

```diff
diff --git a/cmd/openshift-install/create.go b/cmd/openshift-install/create.go
index 7f0ac41..d200632 100644
--- a/cmd/openshift-install/create.go
+++ b/cmd/openshift-install/create.go
@@ -17,6 +17,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 
 	routeclient "github.com/openshift/client-go/route/clientset/versioned"
@@ -91,12 +92,19 @@ var (
 			// FIXME: add longer descriptions for our commands with examples for better UX.
 			// Long:  "",
 			PostRunE: func(_ *cobra.Command, _ []string) error {
-				err := destroyBootstrap(context.Background(), rootOpts.dir)
+				ctx := context.Background()
+
+				config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(rootOpts.dir, "auth", "kubeconfig"))
+				if err != nil {
+					return errors.Wrap(err, "loading kubeconfig")
+				}
+
+				err := destroyBootstrap(ctx, config, rootOpts.dir)
 				if err != nil {
 					return err
 				}
 				logrus.Info("Waiting for openshift-console route to be created...")
-				consoleURL, err := waitForConsole(context.Background(), rootOpts.dir)
+				consoleURL, err := waitForConsole(ctx, config, rootOpts.dir)
 				if err != nil {
 					return err
 				}
@@ -166,20 +174,15 @@ func runTargetCmd(targets ...asset.WritableAsset) func(cmd *cobra.Command, args
 	}
 }
 
-// FIXME: pulling the kubeconfig and metadata out of the root
-// directory is a bit cludgy when we already have them in memory.
-func destroyBootstrap(ctx context.Context, directory string) (err error) {
-	cleanup, err := setupFileHook(rootOpts.dir)
+// FIXME: pulling the metadata out of the root directory is a bit
+// cludgy when we already have it in memory.
+func destroyBootstrap(ctx context.Context, config *rest.Config, directory string) (err error) {
+	cleanup, err := setupFileHook(directory)
 	if err != nil {
 		return errors.Wrap(err, "failed to setup logging hook")
 	}
 	defer cleanup()
 
-	config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(directory, "auth", "kubeconfig"))
-	if err != nil {
-		return errors.Wrap(err, "loading kubeconfig")
-	}
-
 	client, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		return errors.Wrap(err, "creating a Kubernetes client")
@@ -267,19 +270,15 @@ func destroyBootstrap(ctx context.Context, directory string) (err error) {
 	}
 
 	logrus.Info("Destroying the bootstrap resources...")
-	return destroybootstrap.Destroy(rootOpts.dir)
+	return destroybootstrap.Destroy(directory)
 }
 
 // waitForconsole
-func waitForConsole(ctx context.Context, directory string) (string, error) {
+func waitForConsole(ctx context.Context, config *rest.Config, directory string) (string, error) {
 	url := ""
 	// Need to keep these updated if they change
 	consoleNamespace := "openshift-console"
 	consoleRouteName := "console"
-	config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(directory, "auth", "kubeconfig"))
-	if err != nil {
-		return "", errors.Wrap(err, "loading kubeconfig")
-	}
 
 	rc, err := routeclient.NewForConfig(config)
 	if err != nil {
```

on top of your f331f6d.

Reply:
:+1: